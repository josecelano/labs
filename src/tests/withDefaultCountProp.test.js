import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withDefaultCountProp from '../withDefaultCountProp';

const Counter = ({ count }) => (
  <div>Count: {count}</div>
)
const CounterWithDefaultTwelve = withDefaultCountProp(Counter, 12);

it('should set the prop count to `12` by default', () => {
  const tree = toJson(shallow(<CounterWithDefaultTwelve />));
  expect(tree).toMatchSnapshot();
});

it('should accept a prop `count` which should overwrite the default', () => {
  const tree = toJson(shallow(<CounterWithDefaultTwelve count={22} />));
  expect(tree).toMatchSnapshot();
});
