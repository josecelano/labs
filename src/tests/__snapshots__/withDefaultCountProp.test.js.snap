exports[`test should accept a prop \`count\` which should overwrite the default 1`] = `
<Counter
  count={22} />
`;

exports[`test should set the prop count to \`12\` by default 1`] = `
<Counter
  count={12} />
`;
